API: various changes

sunnysideup opened the pull request:
Have a look to see if this makes sense to you at all.

sunnysideup committed:
API: change to different field

sunnysideup committed:
FIX: cleanup references

sunnysideup committed:
FIX: fixnig requirements

sunnysideup committed:
FIX: fixnig requirements

sunnysideup committed:
FIX: code cleanup

sunnysideup committed:
DOC: adding ide annotation

sunnysideup committed:
FIX: better way to deal with anchors

sunnysideup committed:
FIX: check anchor array

sunnysideup committed:
Update SiteTreeLink.php

sunnysideup committed:
FIX: adding getLink for better dev experience

dizzystuff commented:
Hi @sunnysideup thanks for all this! I'm under the gun right now so I'm currently doing as rapid an SS6 upgrade as I'm able to. I'll take a look through and integrate as much as I'm able to today, and then be able to return to do a more thorough job of it in a couple of weeks. Cheers 👍

S-GrabhamMadden committed:
Update composer.json

Change fromholdio/silverstripe-systemlinks from dev-master to supported version, for SS5 functionality

sunnysideup committed:
Merge pull request #1 from S-GrabhamMadden/patch-1

Update composer.json
